### PR TITLE
UPDATE - clean readme, add makefile automation and update gitignore

### PR DESCRIPTION
**Pull request by JohnVonNeumann**
Updates to README. General cleaning. Added a Makefile to start some repository automation and then also updated the gitignore.

Let me know if you want this to be rebased.

```
The following changes since commit a7db1037063b148b6481ee203e9f3c86f6edc443:

  Removing references to whitelist/blacklist. (#215) (2019-03-28 16:22:55 -0700)

are available in the Git repository at:

  git@github.com:JohnVonNeumann/repokid.git add-script-to-provision-base-dynamodb

for you to fetch changes up to e9e8691aaa4d64588a3fd4b779944d384ddc27f9:

  CREATE Makefile - add two new rules: (2019-08-29 17:45:36 +1000)

----------------------------------------------------------------
JohnVonNeumann (3):
      UPDATE gitignore - dont blanket ignore json files, only ignore config.json
      UPDATE README - add content around makefiles
      CREATE Makefile - add two new rules:

 .gitignore |  2 +-
 Makefile   | 19 +++++++++++++++++++
 README.md  |  8 ++++++++
 3 files changed, 28 insertions(+), 1 deletion(-)
 create mode 100644 Makefile

diff --git a/.gitignore b/.gitignore
index e9fff7d..aefc77f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,5 @@
 *.pyc
-*.json
+config.json
 *.log
 *.csv
 build
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..586afc0
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,19 @@
+# Environmental Variables
+PACKAGES=aws
+
+# check_host_system_installs:
+# Runs `which` against program targets to ensure that the users development
+# environment contains all the necessary packages.
+#
+# Environmental Variables:
+#      PACKAGES: A space separated list of packages required in the users
+#                        environment.
+check_host_system_installs:
+       for package in $(PACKAGES); do \
+               which $$package ; \
+       done
+
+# check_aws_region:
+# Retrieves the default AWS region present in your environment.
+check_aws_region:
+       aws configure get default.region
diff --git a/README.md b/README.md
index abf9f5a..b2eb253 100644
--- a/README.md
+++ b/README.md
@@ -146,6 +146,14 @@ $ repokid find_roles_with_permissions "s3:putobjectacl" "sts:assumerole" --outpu
 ...
 $ repokid remove_permissions_from_roles --role-file=myroles.json "s3:putobjectacl" "sts:assumerole" -c

+### Makefile rules
+
+The Makefile at the root of the directory contains some commands that can be used to manage the repository and perform some useful actions that may assist operators/developers.
+
+| command | description |
+| ------- | ----------- |
+| check_host_system_installs | Runs `which` against `binary` targets to check the host level |
+| check_aws_region | Runs an `aws` command that checks the default region set in the local environment. |

```

### CODE REVIEW

**Comment by JohnVonNeumann** [comment]
Yo, life has been busy, did this end up getting the tick on your end? The other PR from what I understand doesn't have a lot to do with this one.

**Comment by JohnVonNeumann** [comment]
Nah all good, I've been busy, as I'm sure you have. I can remove the dynamo creation scripts, now that I look at it, I'm unsure what I was trying to achieve with it haha.

**Comment by mcpeak** [comment]
Sounds good, please keep PR's coming. I have fixed my filter problem so I should actually see them now.

**Comment by JohnVonNeumann** [comment]
Can do, got some stuff I wouldn't mind sinking my teeth into on this project, so when I find some time I'll have a go.

**Comment by JohnVonNeumann** [comment]
@mcpeak Fixed up mate. Removed the cruft. Cheers.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Environmental Variables
+PACKAGES=aws
```
[the remark this answers is not in the document]
Comment (JohnVonNeumann) [reply Makefile]: Ahh sorry, just forgot it, was originally needed for the cli stuff, I'll get this out as well.

**Comment by JohnVonNeumann** [comment]
Burning this branch, scalped so much of the original changes that it's literally got nothing left. I was going to remove the Makefile rule for checking host system installs but just leaving the check_aws_region rule seems like I'm just adding for the sake of adding. I think there's automation to be done on the project, but this isn't it. Sorry for the fuck around.

**Pull request closed without merging**